jimp.d.ts

```diff
@@ -234,369 +234,266 @@ declare namespace Jimp {
+        rotate(
             deg: number,
             mode: number | boolean | undefined | null,
-            cb: Jimp.ImageCallback<T>
-        ): T;
-        flip(horizontal: boolean, vertical: boolean): this;
-        flip<T>(
+            cb?: Jimp.ImageCallback<T>
+        ): this;
+        flip(
             horizontal: boolean,
             vertical: boolean,
-            cb: Jimp.ImageCallback<T>
-        ): T;
-        mirror(horizontal: boolean, vertical: boolean): this;
-        mirror<T>(
+            cb?: Jimp.ImageCallback<T>
+        ): this;
+        mirror(
             horizontal: boolean,
             vertical: boolean,
-            cb: Jimp.ImageCallback<T>
-        ): T;
-        resize<T>(w: number, h: number, cb: Jimp.ImageCallback<T>): T;
-        resize(w: number, h: number, mode?: string): this;
-        resize<T>(
+            cb?: Jimp.ImageCallback<T>
+        ): this;
+        resize(w: number, h: number, cb?: Jimp.ImageCallback<T>): this;
+        resize(
             w: number,
             h: number,
             mode: string | undefined | null,
-            cb: Jimp.ImageCallback<T>
-        ): T;
-        cover<T>(w: number, h: number, cb: Jimp.ImageCallback<T>): T;
-        cover<T>(
+            cb?: Jimp.ImageCallback<T>
+        ): this;
+        cover(w: number, h: number, cb?: Jimp.ImageCallback<T>): this;
+        cover(
             w: number,
             h: number,
             alignBits: number | undefined | null,
-            cb: Jimp.ImageCallback<T>
-        ): T;
-        cover(w: number, h: number, alignBits?: number, mode?: string): this;
-        cover<T>(
+            cb?: Jimp.ImageCallback<T>
+        ): this;
+        cover(
             w: number,
             h: number,
             alignBits: number | undefined | null,
             mode: string | undefined | null,
-            cb: Jimp.ImageCallback<T>
-        ): T;
-        contain(w: number, h: number): this;
-        contain<T>(w: number, h: number, cb: Jimp.ImageCallback<T>): T;
-        contain(w: number, h: number, mode: string): this;
-        contain<T>(
+            cb?: Jimp.ImageCallback<T>
+        ): this;
+        contain(w: number, h: number, cb?: Jimp.ImageCallback<T>): this;
+        contain(
             w: number,
             h: number,
             mode: string,
-            cb: Jimp.ImageCallback<T>
-        ): T;
-        contain(w: number, h: number, alignBits: number): this;
-        contain<T>(
+            cb?: Jimp.ImageCallback<T>
+        ): this;
+        contain(
             w: number,
             h: number,
             alignBits: number,
-            cb: Jimp.ImageCallback<T>
-        ): T;
-        contain(w: number, h: number, alignBits: number, mode: string): this;
-        contain<T>(
+            cb?: Jimp.ImageCallback<T>
+        ): this;
+        contain(
             w: number,
             h: number,
             alignBits: number,

```

Comment:
Both `alignBits` and `mode` should be optional (it's valid to call `contain(20, 20, undefined, undefined, (err, img) => {})`)